neighbors/src/components/RegistrationForm/RegistrationForm.jsx

```diff
@@ -0,0 +1,155 @@
+import '../mainLayout.css';
+
+import { useState } from 'react';
+import { addDoc } from 'firebase/firestore';
+
+import { usersData } from '../../helpers/apiCommunication';
+import { isPasswordValid } from '../../helpers/validation';
+export const Registration = () => {
+  const defaultFormState = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    city: '',
+    street: '',
+    houseNumber: '',
+    phoneNumber: '',
+    password: '',
+  };
+  const [newUser, setNewUser] = useState(defaultFormState);
+  const [error, setError] = useState('');
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    setNewUser({
+      ...newUser,
+      [name]: value,
+    });
+  };
+
+  const onClick = (e) => {
+    e.preventDefault();
+    if (!isPasswordValid(newUser.password)) {
+      setError('Password needs to be at least 6 characters, one special character, and one number!');
+      return;
+    }
+    setError('');
```

Comment:
Przydałaby się walidacja nie tylko na hasło, ale też np. na długość name czy żeby email był emailem.
Warto by także zdefiniować, które pola w formularzu są required, a które optional plus zakomunikować to userowi (np. poprzez gwiazdkę przy obowiązkowych inputach)